src/app/shared/directives/scroll/scroll.directive.ts

```diff
@@ -0,0 +1,17 @@
+import { Directive, HostListener } from '@angular/core';
+
+@Directive({
+  selector: '[appScroll]',
+  standalone: true,
+})
+export class ScrollDirective {
+  @HostListener('click', ['$event'])
+  handleClick(event: Event): void {
+    event.preventDefault();
+
+    const targetId = (event.target as HTMLAnchorElement)?.getAttribute('href')?.substring(1);
```

Comment:
Never cast an element to another type with ```as```. Verify the type instead:
```ts
// This is called an early return if some condition is not met. 
// It's more readable than placing a lot of code into an if 
// statement. (I usually try to avoid indentation as much as I can.
if (event.target instanceOf HTMLAnchorElement === false) {
   return
}
const targetId = event.target.getAttribute('href');
// I use querySelector so I don't have to strip the hashtag from href
const targetElement = targetId && document.querySelector(targetId)
targetElement?.scrollIntoView({ behavior: 'smooth' });
```

Reply:
I would even throw an error to notify the developer that he is missusing the directive.

Reply:
I see, you also want to use it on elements that are not anchors. In that case I'd just declare a property on the directive and assign a value to it.

```ts
@Directive({
  selector: '[appScrollTo]' // Use a selector that suits your naming conventions.
})
export class ScrollToDirective {
  public targetElementId = input<string|null>(null, { alias: 'appScrollTo' })

  @HostListener('click')
  onClick(): void {
    if (!this.targetElementId) {
      return
    }
    document.getElementById(this.targetElementId)?.scrollIntoView({ behavior: 'smooth', block: 'start' });
  }
}
```

Reply:
This early return is really good. I didnt even hear from that before. 😃